### DISCUSSION TITLE
Can you query based on a referred document?

### DISCUSSION THREAD

**Discussion by AlvesJorge** [mongodb/mongoid#5889] [Q&A]
Going with the example schema:
```ruby

class Band
include Mongoid::Document
has_many :awards
field :name, type: String
end
 
class Award
include Mongoid::Document
belongs_to :band
field :prize_money, type: String
field :name, type: String
end

```

I know that if Band embedded Award I could simply query
```ruby
Band.where(award.prize_money: "10k") # [Band1, Band2]
```
But if Band only refers to award how could I perform a similar query? 
This doesn't work:
```ruby
Award.where(prize_money:  "10k", "band.name": "Depeche Mode")
```
**Is there a method that would allow for band to be populated for query purposes?** 
Something like:
```ruby
Award.with_association(:band).where(prize_money:  "10k", "band.name": "Depeche Mode")
```

Using `aggregate` returns BSON documents, which I don't love. 
Embedding awards would stop me from querying for awards directly, which I don't want.
I created an abstraction for this purpose, but I don't love it either, since it requires the knowledge of it's existence.
```ruby
    def self.query_with_band(award_query, band_query)
      awards = Award.where(unit_query)
      band_ids = awards.pluck(:band_id)
      bands = Band.in(id: band_ids).where(band_query)
      award_query.in(band_id: bands.pluck(:id))
    end
```
Which I then call like so
```ruby
Award.query_with_band({prize_money: "10k"}, {name: "Depeche Mode"})
```

TL;DR: Is there a method to allow for querying for refered associations as if they are embedded? :)

PS: I realize that this question may be more geared towards the mongo ruby gem :)

**Comment by jamis** [comment]
Hello @AlvesJorge -- Mongoid does not currently support this style of query; you would need to split it into two queries currently (one to fetch the matching band ids, and the other to find the Award documents for those bands, as you've done in your workaround).

If you're using the driver directly, you can do this with an aggregation pipeline with a ['$lookup'](https://www.mongodb.com/docs/manual/reference/operator/aggregation/lookup/) stage. We plan to add a aggregation pipeline builder DSL to Mongoid "soon", which could likely become a foundation for a feature like what you're describing.

**Reply by AlvesJorge** [reply]
> aggregation pipeline builder DSL to Mongoid "soon", which could likely become a foundation for a feature like what you're describing

I can imagine that being the case :) 
Thank you for the reply jamis
